src/vis/vis.js

```diff
@@ -435,15 +427,23 @@ var Vis = View.extend({
    * Only expected to be called if {skipMapInstantiation} flag is set to true when vis is created.
    */
   instantiateMap: function () {
-    this._instantiateMap();
-    this.mapView.invalidateSize();
-  },
-
-  _instantiateMap: function () {
     this._dataviewsCollection.on('add reset remove', _.debounce(this._invalidateSizeOnDataviewsChanges, 10), this);
     this.map.instantiateMap();
   },
 
+  invalidateMapSize: function () {
+    this.mapView.invalidateSize();
```

Comment:
I don't think we should reset the map to the initial state, it should be "reset" to the current center/zoom, right?